src/Component/tests/Metadata/Resource/Factory/AttributesResourceMetadataCollectionFactoryTest.php

```diff
@@ -581,6 +582,51 @@ public function testItCreatesResourceMetadataWithRoutePrefix(): void
         $this->assertSame('/admin', $operation->getRoutePrefix());
     }
 
+    public function testItCreatesResourceMetadataWithRouteCondition(): void
+    {
+        $this->resourceRegistry
+            ->method('get')
+            ->with('app.dummy')
+            ->willReturn(Metadata::fromAliasAndConfiguration('app.dummy', [
+                'driver' => 'dummy_driver',
+                'classes' => [
+                    'model' => 'App\Dummy',
+                    'form' => 'App\Form',
+                ],
+            ]));
+
+        $metadataCollection = $this->factory->create(DummyResourceWithRouteCondition::class);
+        $this->assertInstanceOf(ResourceMetadataCollection::class, $metadataCollection);
+
+        $resource = $metadataCollection->getIterator()->current();
+        $this->assertInstanceOf(ResourceMetadata::class, $resource);
+        $this->assertSame('app.dummy', $resource->getAlias());
+
+        $operations = $resource->getOperations();
+        $this->assertInstanceOf(Operations::class, $operations);
+        $this->assertCount(4, $operations);
+        $this->assertTrue($operations->has('app_dummy_create'));
```

Comment:
Is it necessary to test resource route condition as well or it's covered by other tests?

Reply:
Yes, you're right. I'll remove those ones.